File: infrapulse_website/frontend/src/types/complaint.ts
export type ComplaintStatusType = 'SUBMITTED' | 'ASSIGNED' | 'IN_PROGRESS' | 'RESOLVED';
export type AIStatusType = 'PENDING' | 'PROCESSING' | 'COMPLETED' | 'FAILED';
export type CategoryType = 'Structural' | 'Functional' | 'Performance';
export type DefectType = 'Cracked_Tiles' | 'Peeling' | 'Spalling' | 'Stagnant_Water';
export type SeverityType = 'LOW' | 'MEDIUM' | 'HIGH';
export type PriorityLevelType = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export interface StatusHistoryResponse {
  id: string;
  complaint_id: string;
  from_status: string;
  to_status: string;
  changed_by_id: string;
  changed_by_role: string;
  changed_at: string;
  notes?: string | null;
}

export interface StatusUpdateRequest {
  status: ComplaintStatusType;
  notes?: string | null;
}

export interface ComplaintResponse {
  id: string;
  user_id: string;
  name_snapshot: string;
  address: string;
  description: string;
  image_url: string;
  
  // AI Results
  ai_status: AIStatusType | string;
  detected_defect?: DefectType | string | null;
  category?: CategoryType | string | null;
  confidence?: number | null;
  
  visible_extent_ratio?: number | null;
  visible_extent_percentage?: number | null;
  extent_label?: string | null;
  extent_score?: number | null;
  
  severity_score?: number | null;
  severity?: SeverityType | string | null;
  
  priority_score?: number | null;
  priority_level?: PriorityLevelType | string | null;
  
  classifier_inference_ms?: number | null;
  pipeline_time_ms?: number | null;
  
  // Status & Timestamps
  status: ComplaintStatusType | string;
  assigned_staff_id?: string | null;
  created_at: string;
  updated_at: string;
  resolved_at?: string | null;
  
  error_code?: string | null;
  error_message?: string | null;
}

export interface ComplaintDetailResponse extends ComplaintResponse {
  queue_position?: number | null;
  queue_size?: number | null;
  status_history: StatusHistoryResponse[];
}

export interface CreateComplaintFormData {
  name: string;
  address: string;
  description: string;
  photo: File;
}

File: infrapulse_website/backend/app/schemas/complaint.py
from typing import List, Optional, Literal
from datetime import datetime
from pydantic import BaseModel, ConfigDict

ComplaintStatusType = Literal["SUBMITTED", "ASSIGNED", "IN_PROGRESS", "RESOLVED"]
AIStatusType = Literal["PENDING", "PROCESSING", "COMPLETED", "FAILED"]
CategoryType = Literal["Structural", "Functional", "Performance"]
DefectType = Literal["Cracked_Tiles", "Peeling", "Spalling", "Stagnant_Water"]
SeverityType = Literal["LOW", "MEDIUM", "HIGH"]
PriorityLevelType = Literal["LOW", "MEDIUM", "HIGH", "CRITICAL"]

class StatusHistoryResponse(BaseModel):
    id: str
    complaint_id: str
    from_status: str
    to_status: str
    changed_by_id: str
    changed_by_role: str
    changed_at: datetime
    notes: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

class StatusUpdateRequest(BaseModel):
    status: ComplaintStatusType
    notes: Optional[str] = None

class ComplaintBase(BaseModel):
    name_snapshot: str
    address: str
    description: str

class ComplaintResponse(BaseModel):
    id: str
    user_id: str
    name_snapshot: str
    address: str
    description: str
    image_url: str
    
    # AI Results
    ai_status: str
    detected_defect: Optional[str] = None
    category: Optional[str] = None
    confidence: Optional[float] = None
    
    visible_extent_ratio: Optional[float] = None
    visible_extent_percentage: Optional[float] = None
    extent_label: Optional[str] = None
    extent_score: Optional[int] = None
    
    severity_score: Optional[float] = None
    severity: Optional[str] = None
    
    priority_score: Optional[float] = None
    priority_level: Optional[str] = None
    
    classifier_inference_ms: Optional[float] = None
    pipeline_time_ms: Optional[float] = None
    
    # Status & Timestamps
    status: str
    assigned_staff_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    resolved_at: Optional[datetime] = None
    
    error_code: Optional[str] = None
    error_message: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

class ComplaintDetailResponse(ComplaintResponse):
    queue_position: Optional[int] = None
    queue_size: Optional[int] = None
    status_history: List[StatusHistoryResponse] = []
